### DISCUSSION TITLE
How to Select All Children Except the `:before` Pseudo-Element in Tailwind CSS?

### DISCUSSION THREAD

**Discussion by MonaAghili** [tailwindlabs/tailwindcss#14872] [Help] [answered]
I’m trying to add a z-index to all child elements within a button except `:before` . However, the code I wrote `[&>*:not(:before)]:z-10` isn’t working as expected.
Also tried `[&>*:not(:where([class*=before]))]:z-10`
BTW, Is there a better solution besides using `z-index`?

https://play.tailwindcss.com/o0cdMxTriQ

**Comment by MonaAghili** [comment]
> I’m trying to add a z-index to all child elements within a button except `:before` . However, the code I wrote `[&>*:not(:before)]:z-10` isn’t working as expected. Also tried `[&>*:not(:where([class*=before]))]:z-10` BTW, Is there a better solution besides using `z-index`?
> 
> https://play.tailwindcss.com/o0cdMxTriQ

My initial approach was incorrect, but [`&>*]:z-10` works if the content is placed within a span. Additionally, `[&>*]:relative` needs to be applied to the elements as well.

PS: I’m curious if there are other solutions for achieving this structure.

**Reply by wongjn** [reply]
You could consider making the `<button>` a [stacking context parent](https://play.tailwindcss.com/o0cdMxTriQ) and then push the `::before` pseudo-element to the back: https://play.tailwindcss.com/o0cdMxTriQ.

**Reply by MonaAghili** [reply]
Well as you can see, the text within button still is under the `:before` pseudo-element
PS: I only changed the color of text to show if it's visible or not.
https://play.tailwindcss.com/MureKOffBP

**Reply by wongjn** [reply, marked as answer]
Yep, that's why I suggested my solution in https://github.com/tailwindlabs/tailwindcss/discussions/14872#discussioncomment-11157499. See the updated version of my solution with your text color change: https://play.tailwindcss.com/GiHe2o7Syb
